application/cms/views.py

```diff
@@ -1074,6 +1059,22 @@ def new_version(topic, subtopic, measure, version):
     )
 
 
+@cms_blueprint.route("/<topic>/<subtopic>/<measure>/<version>/copy", methods=["GET", "POST"])
+@login_required
+@user_can(COPY_MEASURE)
+def copy_measure_page(topic, subtopic, measure, version):
+    topic_page, subtopic_page, measure_page = page_service.get_measure_page_hierarchy(topic, subtopic, measure, version)
+
+    try:
+        page = page_service.create_copy(measure, version, version_type="copy", created_by=current_user.email)
+        message = f"Created a copy of “{measure_page.title}” in subtopic “{subtopic_page.title}”"
+        flash(message, "info")
+        return redirect(url_for("static_site.topic", uri=topic_page.uri))
+    except Exception as e:
+        # TODO: Figure out what errors need to be handled
```

Comment:
I'd expect a failing copy operation to redirect back to the edit form for the measure you were copying, displaying an error as to what went wrong?

That said, the only reasons for a fail should be authentication & authorization, which are already handled by Flask, so maybe we don't need to catch any exceptions here?

Reply:
You never know... database might randomly say no, copying uploads fails, etc. Best to handle and tell the user.